src/software/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -107,3 +109,37 @@ std::set<RobotCapability> &Tactic::mutableRobotCapabilityRequirements()
 {
     return capability_reqs;
 }
+
+std::unique_ptr<Intent> Tactic::get(const Robot &robot, const World &world)
+{
+    // TODO (#1888): remove these updates
+    updateWorldParams(world);
+    updateRobot(robot);
+
+    updateIntent(TacticUpdate{.robot      = robot,
+                              .world      = world,
+                              .set_intent = [this](std::unique_ptr<Intent> new_intent) {
+                                  intent = std::move(new_intent);
+                              }});
```

Comment:
why does this function need to be dependency injected on every get?

is this mostly to capture "this" tactic and set the intent?

Reply:
Not sure what you mean, set_intent should be a member of Tactic?